lambda/rest-endpoints/src/helpers.test.ts

```diff
@@ -41,4 +47,62 @@ describe("app.validateDateParam", ()=>{
     ).toThrow("Invalid year");
   });
 
+});
+
+describe("app.recursivelyGetIdList", ()=>{
+  beforeEach(()=>{
+    jest.resetAllMocks();
+  });
+
+  it("should batch up requests and send to the dynamo layer", async ()=>{
+    (multipleRecipesByUid as jest.Mock).mockImplementation((idList:string[])=>{
+      return idList.map((uuid)=>({
+        checksum: `cs-for-${uuid}`,
+        recipeUID: uuid,
+        capiArticleId: `article-for-${uuid}`,
+        sponsorshipCount: 0
+      }))
+    });
+
+    const uidList:string[] = [];
+
+    for(let i=0; i<120; i++) {
+      uidList.push(uuid());
+    }
+
+    const results = await recursivelyGetIdList(uidList, []);
+    expect(results.length).toEqual(120);
+    expect(results.map(e=>e.recipeUID)).toEqual(uidList);
+    //we expect there to have been 3 parallel batches
+    expect((multipleRecipesByUid as jest.Mock).mock.calls.length).toEqual(3);
+  });
+
+  it("should retry if ProvisionedThroughputExceeded is caught", async ()=>{
+    (multipleRecipesByUid as jest.Mock).mockImplementation((idList:string[])=>{
+      const shouldCrash = Math.random();
```

Comment:
Is this `Math.random`isation reliable enough?

Reply:
Well, the idea is that we crash ~5% of the time - on 120 resolves that should average 6 times so we'd have to be _really_ unlucky not to trigger at least one.  The test won't actually fail in this scenario; it's just extremely unlikely to happen